tests/unit/core/middleware/test_cspMiddleware.js

```diff
@@ -5,15 +5,52 @@ import parse from 'content-security-policy-parser';
 import { csp } from 'core/middleware';
 import { getFakeConfig, getFakeLogger } from 'tests/unit/helpers';
 
+const deployedEnvs = ['dev', 'production', 'stage'];
+
+const cdnHosts = {
+  dev: 'https://addons-dev-cdn.allizom.org',
+  stage: 'https://addons-stage-cdn.allizom.org',
+  production: 'https://addons.cdn.mozilla.net',
+};
+
+const apiHosts = {
+  dev: 'https://addons-dev.allizom.org',
+  stage: 'https://addons.allizom.org',
+  production: 'https://addons.mozilla.org',
+};
+
 describe(__filename, () => {
-  describe('CSP Middleware', () => {
-    const existingNodeEnv = process.env.NODE_ENV;
+  const existingNodeEnv = process.env.NODE_ENV;
+
+  afterEach(() => {
+    process.env.NODE_ENV = existingNodeEnv;
+    delete process.env.NODE_APP_INSTANCE;
```

Comment:
I know this is how the old tests did it but can you make this preserve the old value? Something like:

```js
const existingNodeAppInstance = process.env.NODE_APP_INSTANCE;
...
afterEach(() => {
  ...
  process.env.NODE_APP_INSTANCE = existingNodeAppInstance;
});
```

Reply:
I've tried the following, which passes the tests, but doesn't make sense to me:

```js
  const existingNodeEnv = process.env.NODE_ENV;
  const existingNodeAppInstance = process.env.NODE_APP_INSTANCE;

  afterEach(() => {
    process.env.NODE_ENV = existingNodeEnv;
    process.env.NODE_APP_INSTANCE = existingNodeAppInstance;
    delete process.env.NODE_APP_INSTANCE;
  });
```

`process.env.NODE_APP_INSTANCE = existingNodeAppInstance;` is redundant since the `delete ` statement removes it from [process.env](https://nodejs.org/api/process.html#process_process_env). Without that line, most of the tests fail which leaves me scratching my head.

We want to essentially reset `NODE_APP_INSTANCE` to what it was before the tests ran - right?

Reply:
I slapped in those changes and cleaned up the tests by adding separate `describe` blocks where necessary. Hopefully this just needs one more review @kumar303!